File: c/LinkedListImplementation.c
#include <stdio.h> 
#include <stdlib.h> 

struct Node { 
	int data; 
	struct Node* next; 
}; 

struct Node* head;

void add(int data){
    struct Node* toAdd = (struct Node*)malloc(sizeof(struct Node)); 
    toAdd->data = data;
    toAdd->next = NULL;

    if(head == NULL){
        head = toAdd;
        return;
    }
    struct Node* temp = head;

    while(temp->next != NULL){
        temp = temp->next;
    }
    temp->next = toAdd;
}

void printAll()  { 
    struct Node* temp = head;

    while (temp != NULL) { 
        printf("%d ", temp->data); 
        temp = temp->next; 
    } 
} 

void search(int data){
    struct Node* temp = head;

    while (temp != NULL) { 
        if(temp->data == data){
            printf("Element Present");
            return;
        }
        temp = temp->next; 
    }
    printf("Element Not Present");
}

int main()  { 

    add(12);
    add(1);
    add(10);

    printAll();
    
    search(12);

	return 0; 
} 

File: java/LinkedlistImplementation.java
public class LinkedlistImplementation {
	
	public class Node{
		int data;
		Node next;
		
		Node(int data){
			this.data = data;
			next = null;
		}
	}
	
	Node head;
	
	boolean isEmpty() {
		return head == null;
	}
	
	void add(int data) {
		Node toAdd = new Node(data);
		
		if(isEmpty()) {
			head = toAdd;
			return;
		}
		
		Node temp = head;
		while(temp.next != null) {
			temp = temp.next;
		}
		temp.next = toAdd;
	}
	
	void printAll() {
        if(isEmpty()){
            System.out.println("Empty List");
            return;
        }
		Node temp = head;
		while(temp != null) {
			System.out.println(temp.data);
			temp = temp.next;
		}
	}

    void search(int data) {
		if(isEmpty()) {
			System.out.println("Empty List");
			return;
		}
		
		Node temp = head;
		while(temp != null) {
			if(temp.data == data) {
				System.out.println("Found");
				return;
			}
			temp = temp.next;
		}
		
		System.out.println("Not Found");
    }
    
	public static void main(String[] args) {
		LinkedlistImplementation ll = new LinkedlistImplementation();
		
		ll.add(12);
		ll.add(1);
		ll.add(22);
		ll.add(10);
		
        ll.printAll();
        
        ll.search(22);
	}

}
